cabal-install/main/Main.hs

```diff
@@ -907,10 +908,16 @@ unpackAction getFlags extraArgs globalFlags = do
   getAction getFlags extraArgs globalFlags
 
 initAction :: InitFlags -> [String] -> Action
-initAction initFlags extraArgs globalFlags
-    | not (null extraArgs) =
-      die' verbosity $ "'init' doesn't take any extra arguments: " ++ unwords extraArgs
-    | otherwise = do
+initAction initFlags extraArgs globalFlags = do
+  -- it takes the first value within extraArgs (if there's one)
+  -- and uses it as the root directory for the new project
+  case nonEmpty extraArgs of
+    Just (projectDir :| []) -> do
+      createDirectoryIfMissing True projectDir
+      withCurrentDirectory projectDir initAction'
+    _ -> initAction'
```

Comment:
This silently ignores all arguments if there isn't exactly one. Instead I think it would be better behavior for it to only init in the current directory on zero arguments, and die with an error on two or more arguments.